pkg/util/merge/merge.go

```diff
@@ -0,0 +1,181 @@
+		merged.TerminationMessagePolicy = overrideContainer.TerminationMessagePolicy
+	}
+
+	if overrideContainer.ImagePullPolicy != "" {
+		merged.ImagePullPolicy = overrideContainer.ImagePullPolicy
+	}
+
+	// TODO: merge SecurityContext
+
+	if overrideContainer.Stdin {
+		merged.Stdin = overrideContainer.Stdin
+	}
+
+	if overrideContainer.StdinOnce {
+		merged.StdinOnce = overrideContainer.StdinOnce
+	}
+
+	if overrideContainer.TTY {
+		merged.TTY = overrideContainer.TTY
+	}
+
+	return merged
+}
+
+// ContainerPorts merges all of the fields of the overridePort on top of the defaultPort
+// if the fields don't have a zero value. Thw new ContainerPort is returned.
+func ContainerPorts(defaultPort, overridePort corev1.ContainerPort) corev1.ContainerPort {
+	mergedPort := defaultPort
+	if overridePort.Name != "" {
+		mergedPort.Name = overridePort.Name
+	}
+	if overridePort.ContainerPort != 0 {
+		mergedPort.ContainerPort = overridePort.ContainerPort
+	}
+	if overridePort.HostPort != 0 {
+		mergedPort.HostPort = overridePort.HostPort
+	}
+	if overridePort.Protocol != "" {
+		mergedPort.Protocol = overridePort.Protocol
+	}
+	if overridePort.HostIP != "" {
+		mergedPort.HostIP = overridePort.HostIP
+	}
+	return mergedPort
+}
+
+// ContainerPortSlicesByName takes two slices of corev1.ContainerPorts, these values are merged by name.
+// if there are elements present in the overridePorts that are not present in defaultPorts, they are
+// appended to the end.
+func ContainerPortSlicesByName(defaultPorts, overridePorts []corev1.ContainerPort) []corev1.ContainerPort {
+	defaultPortMap := createContainerPortMap(defaultPorts)
+	overridePortsMap := createContainerPortMap(overridePorts)
+
+	mergedPorts := make([]corev1.ContainerPort, 0)
+
+	for portName, defaultPort := range defaultPortMap {
+		if overridePort, ok := overridePortsMap[portName]; ok {
+			mergedPorts = append(mergedPorts, ContainerPorts(defaultPort, overridePort))
+		} else {
+			mergedPorts = append(mergedPorts, defaultPort)
+		}
+	}
+
+	for portName, overridePort := range overridePortsMap {
+		if _, ok := defaultPortMap[portName]; !ok {
+			mergedPorts = append(mergedPorts, overridePort)
+		}
+	}
+
+	sort.SliceStable(mergedPorts, func(i, j int) bool {
+		return mergedPorts[i].Name < mergedPorts[j].Name
+	})
+
+	return mergedPorts
+}
+
+func createContainerPortMap(containerPorts []corev1.ContainerPort) map[string]corev1.ContainerPort {
+	containerPortMap := make(map[string]corev1.ContainerPort)
+	for _, m := range containerPorts {
+		containerPortMap[m.Name] = m
+	}
+	return containerPortMap
+}
+
+// VolumeMount merges two corev1.VolumeMounts. Any fields in the override take precedence
+// over the values in the original. Thw new VolumeMount is returned.
+func VolumeMount(original, override corev1.VolumeMount) corev1.VolumeMount {
+	merged := original
+
+	if override.Name != "" {
+		merged.Name = override.Name
+	}
+
+	if override.ReadOnly {
+		merged.ReadOnly = override.ReadOnly
+	}
```

Comment:
[q] does this mean we can only override with `true` and never with `false`?

Reply:
yes, unfortunately it doesn't look like there is an easy way to allow overrides for falsy values. There is no way to distinguish between a specification of `false` versus the absence of a value.